### PR TITLE
Redirect to feed after completing the bank connection

### PR DESCRIPTION
**Pull request by DylanDylann**
### Fixed Issues

$ https://expensify.slack.com/archives/C06ML6X0W9L/p1730461357005019?thread_ts=1729596921.819569&cid=C06ML6X0W9L
PROPOSAL: https://expensify.slack.com/archives/C06ML6X0W9L/p1730688809259139?thread_ts=1729596921.819569&cid=C06ML6X0W9L

### Tests

- [x] Verify that no errors appear in the JS console

### Internal QA Steps

1. Go to Company Card Page
2. Add new Company Card
3. Select Chase Bank
4. Select Direct Feed
5. Complete the bank connection
6. Verify that the App navigate to Company Feed Page
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/5d780c3b-11b8-4404-8641-83ba73172b25

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/f18abd35-de4d-4489-bd8a-e203b09cab2f

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/ae0d9123-a58c-4762-95e5-f3ed92178006

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/340cf561-2cf4-49b1-a688-51f09515c6fd

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/d6a135c1-13f9-4036-944c-9b30624d9c98

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/4dce22b8-ca4a-4f3a-918e-7d050771fb6d

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@mountiny The PR is up

**Comment by DylanDylann** [comment]
@dukenv0307 @allgandalf Could you guys help to review this PR? Please comment on this PR and start reviewing soon. Thanks

**Comment by DylanDylann** [comment]
Please comment on the code change then the Melvin will not get another C+ when I open the PR

**Comment by DylanDylann** [comment]
@dukenv0307 I still trying to test this change. It seems hard to test on dev without an auth connection

**Comment by DylanDylann** [comment]
@dukenv0307 Problem: When users complete the bank account flow, the App doesn't navigate them to Company Feed. This PR will fix this problem


https://github.com/user-attachments/assets/deb3fc7b-132a-46cf-a9cf-81fd13ba2aef

**Comment by mountiny** [comment]
I anticipate this is hard to test locally, sorry there is no way to handle it.

How does the xero connection or other integration connections handle it, i think they must use the same approach

File: src/pages/workspace/companyCards/addNew/AddNewCardPage.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ function AddNewCardPage({policy}: WithPolicyAndFullscreenLoadingProps) {
             case CONST.COMPANY_CARDS.STEP.CARD_TYPE:
                 return <CardTypeStep />;
             case CONST.COMPANY_CARDS.STEP.BANK_CONNECTION:
-                return <BankConnection policyID={policyID} />;
+                return <BankConnection policyID={policyID ?? '-1'} />;
```
Comment (DylanDylann) [inline src/pages/workspace/companyCards/addNew/AddNewCardPage.tsx:33]:
```suggestion
                return <BankConnection policyID={policyID} />;
```

File: src/pages/workspace/companyCards/addNew/BankConnection/index.tsx [hunk 2]
```diff
@@ -20,7 +21,7 @@ import openBankConnection from './openBankConnection';
 let customWindow: Window | null = null;
 
 type BankConnectionStepProps = {
-    policyID?: string;
+    policyID: string;
```
Comment (DylanDylann) [inline src/pages/workspace/companyCards/addNew/BankConnection/index.tsx:24]:
```suggestion
    policyID?: string;
```

**Comment by DylanDylann** [comment]
> I anticipate this is hard to test locally, sorry there is no way to handle it.
How does the xero connection or other integration connections handle it, i think they must use the same approach

This is another flow, and I don't think we can use the same way here. Trying another way

**Comment by DylanDylann** [comment]
Mock data to test on dev env:


On src/pages/workspace/companyCards/addNew/BankConnection/index.tsx

Add
```
    const mockBankResponse = () => {
        Onyx.set(`${ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_DOMAIN_MEMBER}${18600095}`, {       // PLease update your workspace account id
            settings: {
                accountManager: 9666036,
                accountManagerAssignmentDate: '2023-11-02 14:17:44',
                companyCardNicknames: {'ofx.chase.com': ''},
                companyCards: {
                    'Expensify Card': {
                        asrEnabled: false,
                        forceReimbursable: 'force_no',
                        liabilityType: 'personal',
                        preferredPolicy: '',
                        reportTitleFormat: '',
                        statementPeriodEndDay: 1,
                        uploadLayoutSettings: null,
                    },
                    'oauth.chase.com': {
                        asrEnabled: false,
                        forceReimbursable: 'force_no',
                        liabilityType: 'personal',
                        preferredPolicy: '',
                        reportTitleFormat: '',
                        statementPeriodEndDay: 1,
                        uploadLayoutSettings: null,
                    },
                },
                oAuthAccountDetails: {
                    'oauth.chase.com': {
                        accountList: ['C. DUNBAR...0546', 'J. HAYES...4754', 'K. BAACK...8211', 'L. BORGNINO...9008', 'C. MILES...4129', 'J. BRUSH...8303'],
                        credentials: 'ASDF',
                        expiration: 1730127183,
                    },
                },
                technicalContactEmail: 'brad.sandt@k12itc.com',
                twoFactorAuthExemptEmails: ['christi.orem@k12itc.com'],
                useTechnicalContactBillingCard: true,
            },
        });
    };


```

Add a button to trigger mock data
```
            <Button onPress={mockBankResponse}>
                <Text>Click me</Text>
            </Button>
```

**Comment by DylanDylann** [comment]
@mountiny For sure, we need to make a build and reach out to some internal QA who have chase accounts to test this PR

**Comment by DylanDylann** [comment]
@dukenv0307 Could you help to test this PR on other platforms?

**Comment by DylanDylann** [comment]
@dukenv0307 On native, I still get some problems

**Comment by DylanDylann** [comment]
There are some updates

We only need to mock this response

```
        Onyx.set(`${ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_DOMAIN_MEMBER}${18600095}`, {
            settings: {
                oAuthAccountDetails: {
                    'oauth.chase.com': {
                        accountList: ['C. DUNBAR...0546', 'J. HAYES...4754', 'K. BAACK...8211', 'L. BORGNINO...9008', 'C. MILES...4129', 'J. BRUSH...8303'],
                        credentials: 'ASDF',
                        expiration: 1730127183,
                    },
                },
            },
        });
```

**Comment by DylanDylann** [comment]
@dukenv0307 Please help to review again

File: src/types/onyx/CardFeeds.ts [hunk 3]
```diff
@@ -61,6 +61,12 @@ type CardFeeds = {
 
                 /** Expiration number */
                 expiration: number;
+
+                /** Whether any actions are pending */
+                pending?: boolean;
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/types/onyx/CardFeeds.ts:66]: pending could be undefined

File: src/pages/workspace/companyCards/addNew/BankConnection/index.native.tsx [hunk 4]
```diff
@@ -46,6 +56,16 @@ function BankConnection({policyID}: BankConnectionStepProps) {
         setWebViewOpen(true);
     }, []);
 
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/pages/workspace/companyCards/addNew/BankConnection/index.native.tsx:59]: I think it is unnecessary, the code isn't the same totally and it also is small

**Comment by DylanDylann** [comment]
@dukenv0307 kindly bump

**Review by dukenv0307 — APPROVED** [review]
LGTM

**Review by mountiny — APPROVED** [review]
Changes look good to me, we will continue improving this code overtime but right now mainly we want to get the connection working smoothly so we can test other flows easier

@JmillsExpensify will help us test this

**Review by dukenv0307 — APPROVED** [review]

**Pull request merged**